src/vs/workbench/api/node/mainThreadEditors.ts

```diff
@@ -102,11 +104,16 @@ export class MainThreadEditors extends MainThreadEditorsShape {
 
 	// --- from extension host process
 
-	$tryShowTextDocument(resource: URI, position: EditorPosition, preserveFocus: boolean): TPromise<string> {
+	$tryShowTextDocument(resource: URI, options: vscode.ShowTextDocumentOptions): TPromise<string> {
+		const position: EditorPosition = TypeConverters.fromViewColumn(options.column);
```

Comment:
Don't use the TypeConverter here but send proper data. Do all the massaging before calling a main-side method. Rule of thumb is that `mainXYZ` files shouldn't depend on `extHostXZY` files